server/src/main/java/org/opensearch/index/compositeindex/StarTreeFieldSpec.java

```diff
@@ -0,0 +1,66 @@
+/*
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * The OpenSearch Contributors require contributions made to
+ * this file be licensed under the Apache-2.0 license or a
+ * compatible open source license.
+ */
+
+package org.opensearch.index.compositeindex;
+
+import org.opensearch.common.annotation.ExperimentalApi;
+
+import java.util.List;
+import java.util.Locale;
+import java.util.concurrent.atomic.AtomicInteger;
+
+/**
+ * Star tree index specific settings for a composite field.
+ */
+@ExperimentalApi
+public class StarTreeFieldSpec implements CompositeFieldSpec {
+
+    private final AtomicInteger maxLeafDocs = new AtomicInteger();
+    private final List<String> skipStarNodeCreationInDims;
```

Comment:
I guess we will need to expose these fields via getters so that we can skip these during star tree creation. Build Mode will help us switch between onheap and offheap modes